asdf/generic_io.py

```diff
@@ -370,7 +369,20 @@ def write(self, content):
     """
 
     def write_array(self, array):
-        _array_tofile(None, self.write, array.ravel(order='K'))
+        """
+        Write array content to the file.  Array must be 1D contiguous
+        so that this method can avoid making assumptions about the
+        intended memory layout.  Endianness is preserved.
+
+        Parameters
+        ----------
+        array : np.ndarray
+            Must be 1D contiguous.
+        """
+        if len(array.shape) != 1 or not array.flags.contiguous:
+            raise ValueError("Requires 1D contiguous array.")
+
+        _array_tofile(None, self.write, array)
 
     def seek(self, offset, whence=0):
```

Comment:
This is the first time I've seen `whence` as a kwarg.  I like.  Though it's not really clear from the docstring what it does.

Reply:
That's funny, I didn't notice that.  Seems to be a [bit of a tradition](https://www.gnu.org/software/libc/manual/html_node/File-Positioning.html#index-fseek).

Reply:
😅 Shows you how much I paid attention in my C programming!